source/appController/AMDSClientAppController.cpp

```diff
@@ -0,0 +1,199 @@
+#include "AMDSClientAppController.h"
+
+#include <QtNetwork>
+
+#include "source/ClientRequest/AMDSClientRequest.h"
+#include "source/ClientRequest/AMDSClientIntrospectionRequest.h"
+#include "source/Connection/AMDSClientTCPSocket.h"
+#include "source/Connection/AMDSServer.h"
+#include "source/util/AMDSErrorMonitor.h"
+
+
+AMDSClientAppController::AMDSClientAppController(QObject *parent) :
+	QObject(parent)
+{
+	networkSession_ = 0;
+}
+
+AMDSClientAppController::~AMDSClientAppController()
+{
+	// release the resouces of the AMDSServers
+	foreach (QString key, activeServers_.keys()) {
+		AMDSServer *server = activeServers_.value(key);
+		if (server)
+			server->deleteLater();
+
+		activeServers_.remove(key);
+	}
+
+	if (networkSession_)
+		networkSession_->deleteLater();
+}
+
+bool AMDSClientAppController::isSessionOpen()
+{
+	return (!networkSession_ || networkSession_->isOpen());
```

Comment:
I believe this should actually be `return (networkSession_ && networkSession_->isOpen();`